x-pack/legacy/plugins/ml/public/application/explorer/explorer.js

```diff
@@ -217,717 +164,119 @@ export const Explorer = injectI18n(injectObservablesAsProps(
-                    {
-                      defaultMessage:
-                  'Filter by influencer fields… ({queryExample})',
-                      values: {
-                        queryExample:
-                    `${influencerName} : ${stateUpdate.influencers[influencerName][0].influencerFieldValue}`
-                      }
-                    }
-                  ));
-            break;
-          }
-        }
-      }
-
-      const updatedAnomalyChartRecords = await loadDataForCharts(
-        jobIds, timerange.earliestMs, timerange.latestMs, selectionInfluencers, selectedCells, influencersFilterQuery
-      );
-
-      if ((selectionInfluencers.length > 0 || influencersFilterQuery !== undefined) && updatedAnomalyChartRecords !== undefined) {
-        stateUpdate.influencers = await getFilteredTopInfluencers(
-          jobIds,
-          timerange.earliestMs,
-          timerange.latestMs,
-          updatedAnomalyChartRecords,
-          selectionInfluencers,
-          noInfluencersConfigured,
-          influencersFilterQuery
-        );
-      }
-
-      stateUpdate.anomalyChartRecords = updatedAnomalyChartRecords || [];
-
-      this.setState(stateUpdate);
-
-      if (selectedCells !== null) {
-        this.updateCharts(
-          stateUpdate.anomalyChartRecords, timerange.earliestMs, timerange.latestMs
-        );
-      } else {
-        this.updateCharts(
-          [], timerange.earliestMs, timerange.latestMs
-        );
-      }
+    componentWillUnmount() {
+      this._unsubscribeAll.next();
+      this._unsubscribeAll.complete();
+      this.resizeChecker.destroy();
+    }
 
-      const { tableInterval, tableSeverity } = this.props;
-      const anomaliesTableCompareArgs = {
-        selectedCells,
-        selectedJobs,
-        dateFormatTz,
-        interval: this.getSwimlaneBucketInterval(selectedJobs).asSeconds(),
-        boundsMin: bounds.min.valueOf(),
-        boundsMax: bounds.max.valueOf(),
-        swimlaneViewByFieldName: viewBySwimlaneOptions.swimlaneViewByFieldName,
-        tableInterval,
-        tableSeverity,
-        influencersFilterQuery
-      };
+    resetCache() {
+      this.anomaliesTablePreviousArgs = null;
+    }
 
-      if (_.isEqual(anomaliesTableCompareArgs, this.anomaliesTablePreviousArgs)) {
-        this.setState(this.anomaliesTablePreviousData);
-      } else {
-        this.anomaliesTablePreviousArgs = anomaliesTableCompareArgs;
-        const tableData = this.anomaliesTablePreviousData = await loadAnomaliesTableData(
-          selectedCells,
-          selectedJobs,
-          dateFormatTz,
-          this.getSwimlaneBucketInterval(selectedJobs).asSeconds(),
-          bounds,
-          viewBySwimlaneOptions.swimlaneViewByFieldName,
-          tableInterval,
-          tableSeverity,
-          influencersFilterQuery
-        );
-        this.setState({ tableData });
+    componentDidUpdate() {
+      // TODO migrate annotations update
```

Comment:
Is this TODO needed in this PR?

Reply:
This is meant for a follow up (possibly should we move this component to use hooks and `useObservable()`. I removed the comment `// this.updateExplorer();` in one of the next lines though in d358c3c9e6.